paddle/fluid/operators/elementwise/mkldnn/elementwise_add_mkldnn_op.cc

```diff
@@ -33,27 +33,45 @@ class EltwiseAddMKLDNNGradKernel : public ElemwiseGradKernel<T> {
     ElemwiseGradKernel<T>::Compute(ctx);
     using Tensor = framework::Tensor;
 
+    auto& dev_ctx =
+        ctx.template device_context<paddle::platform::MKLDNNDeviceContext>();
+    const auto& onednn_engine = dev_ctx.GetEngine();
+
     auto* dout = ctx.Input<Tensor>(framework::GradVarName("Out"));
     auto* dx = ctx.Output<Tensor>(framework::GradVarName("X"));
     auto* dy = ctx.Output<Tensor>(framework::GradVarName("Y"));
 
-    auto set_mkldnn_format = [](Tensor* in, const Tensor* out) {
-      in->set_layout(DataLayout::kMKLDNN);
-      in->set_format(out->format());
-    };
+    auto tz = paddle::framework::vectorize<int64_t>(dout->dims());
+    memory::data_type dout_type = framework::ToMKLDNNDataType(dout->type());
+    std::string key = platform::CreateKey(dev_ctx, tz, dout->format(),
+                                          dout->format(), dout_type);
+    platform::ReorderMKLDNNHandler handler(tz, dout->type(), dout_type, dev_ctx,
+                                           onednn_engine, key);
+
+    mkldnn::stream astream(onednn_engine);
+    auto reorder_src_memory_p = handler.AcquireSrcMemory(
+        dout->format(), platform::to_void_cast(dout->data<T>()));
 
-    // TODO(jczaja): Double check if vcopy works for blocked data
-    auto blas = math::GetBlas<paddle::platform::CPUDeviceContext, T>(ctx);
     if (dx) {
-      blas.VCOPY(dout->numel(), dout->data<T>(),
-                 dx->mutable_data<T>(ctx.GetPlace()));
-      set_mkldnn_format(dx, dout);
+      auto reorder_dst_memory_p =
+          handler.AcquireDstMemory(dx, dout->format(), ctx.GetPlace());
+      auto reorder_p =
+          handler.AcquireReorder(reorder_dst_memory_p, reorder_src_memory_p);
+      platform::RecordEvent record_reorder("int_reorder",
+                                           platform::EventRole::kUniqueOp);
+      reorder_p->execute(astream, *reorder_src_memory_p, *reorder_dst_memory_p);
+      astream.wait();
     }
 
     if (dy) {
```

Comment:
If both guards are not needed maybe we can squash both dx, dy reorders into one

Reply:
This op may be executed when dx is null and dy is present or other way around. Not sure if it can (makes sense?) to execute it without any propagation .This is why I left it the way I found it e.g. two separate blocks if (dx).. and if (dy).. If you know how without changing logic to make it more maintainable and nicer looking then let me know